lib/pages/faq/view/faq_page.dart

```diff
@@ -0,0 +1,176 @@
+                  child: Selectable(
+                    label: category,
+                    initiallySelected: false,
+                    onSelected: (selection) {
+                      setState(() {
+                        if (selection) {
+                          activeTags.add(category);
+                        } else {
+                          activeTags.remove(category);
+                        }
+                      });
+                    },
+                  ),
+                ))
+            .toList(),
+      );
+
+  @override
+  Widget build(BuildContext context) {
+    return AppScaffold(
+      title: S.of(context).sectionFAQ,
+      actions: [
+        AppScaffoldAction(
+          icon: Icons.search,
+          onPressed: () {
+            setState(() {
+              searchClosed = !searchClosed;
+            });
+          },
+        )
+      ],
+      body: FutureBuilder(
+          future: QuestionsService().getDocuments(),
+          builder: (context, snapshot) {
+            if (!snapshot.hasData) return LinearProgressIndicator();
+            questions = QuestionsService().getQuestions(snapshot.data);
+            categories = questions.expand((e) => e.tags).toSet().toList();
+            return ListView(
+              children: [
+                SearchWidget(
+                  title: categoryList(),
+                  onSearch: (searchText) {
+                    setState(() {
+                      filter = searchText;
+                    });
+                  },
+                  cancelCallback: () {
+                    setState(() {
+                      searchClosed = true;
+                      filter = '';
+                    });
+                  },
+                  searchClosed: searchClosed,
+                ),
+                QuestionsList(
+                    questions: questions
+                        .where((question) =>
+                            filter
+                                .split(' ')
+                                .where((element) => element != '')
+                                .fold(
+                                    true,
+                                    (previousValue, filter) =>
+                                        previousValue &&
+                                        question.question
+                                            .toLowerCase()
+                                            .contains(filter)) &&
+                            (activeTags.isEmpty
+                                ? true
+                                : containsTag(activeTags, question.tags)))
+                        .toList(),
+                    filter: filter),
+              ],
+            );
+          }),
+    );
+  }
+
+  bool containsTag(List<String> activeTags, List<String> questionTags) {
+    for (String tag in activeTags) {
+      if (questionTags.contains(tag)) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
This whole thing can be something like `questionTag.any(activeTags.contains)`.

https://dartpad.dartlang.org/b51c02cd3050aa77617943dbe8ab7a8b